contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -1251,6 +1272,58 @@ SendColumnMetadataToken(int natts, bool sendRowStat)
 			}
 		}
 
+		/* For Spatial Types */
```

Comment:
Is this section supposed to come after table name? Please test with a cursor. IIRC, cursor response sends a table name token. 

Anyway, we shouldn't write this code like this. Introduce the following field in TdsColumnMetaData
`
StringInfoData fully_qualified_type_name;
`
In SetColMetadataForGeometryType and SetColMetadataForGeographyType, initialize this field. No need to reuse SetColMetadataForBinaryType. I believe the type deserves its own metadata function.

In SendColumnMetadataToken, just do
`
If col->fully_qualified_type_name != NULL and col->fully_qualified_type_name.len > 0
TdsPutbytes(col->fully_qualified_type_name.data, col->fully_qualified_type_name.len)
`

Reply:
Done